src/analytics/mod.rs

```diff
@@ -3,5 +3,247 @@
 
 //! Various analytics that give insight into the usage of the tangle.
 
+use futures::TryStreamExt;
+use thiserror::Error;
+
+use self::{
+    ledger::{
+        AddressActivity, AddressBalanceAnalytics, AliasActivityAnalytics, BaseTokenActivityAnalytics,
+        LedgerOutputAnalytics, LedgerSizeAnalytics, NftActivityAnalytics, TransactionAnalytics,
+        UnclaimedTokenAnalytics,
+    },
+    tangle::{BlockActivityAnalytics, BlockAnalytics, MilestoneSizeAnalytics},
+};
+use crate::{
+    db::{
+        collections::analytics::{Measurement, OutputActivityAnalyticsResult, PerMilestone, TimeInterval},
+        influxdb::InfluxDb,
+    },
+    tangle::{BlockData, InputSource, LedgerUpdateStore, Milestone},
+    types::{
+        stardust::block::{payload::TransactionEssence, Input, Payload},
+        tangle::{MilestoneIndex, ProtocolParameters},
+    },
+};
+
 pub mod ledger;
 pub mod tangle;
+
+#[allow(missing_docs)]
+pub enum Analytic {
+    AddressBalance(AddressBalanceAnalytics),
+    BaseTokenActivity(BaseTokenActivityAnalytics),
+    BlockActivity(BlockActivityAnalytics),
+    DailyActiveAddresses(AddressActivity),
+    LedgerOutputs(LedgerOutputAnalytics),
+    LedgerSize(LedgerSizeAnalytics),
+    MilestoneSize(MilestoneSizeAnalytics),
+    OutputActivity {
+        nft: NftActivityAnalytics,
+        alias: AliasActivityAnalytics,
```

Comment:
Now that we have in-memory analytics, we should be also able to compute `FoundryActivityAnalytics` (which we could not do before). Feel free to open an issue if you don't want to do it in this PR though.

Reply:
#1081